compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2104,6 +2106,16 @@ impl<'tcx> Place<'tcx> {
         self.projection.iter().any(|elem| elem.is_indirect())
     }
 
+    /// If MirPhase >= Derefered and if projection contains Deref,
+    /// It's guaranteed to be in the first place
+    pub fn ret_deref(&self) -> Option<PlaceElem<'tcx>> {
```

Comment:
These two functions are only ever used to check for `is_some()`, change the return type to a `bool` and change the name to `needs_deref` (name bikesheddable, probably not ideal, if you have suggestions, please lmk)